### ISSUE TITLE
wrong-import-order from running as a subprocess from another venv

### ISSUE THREAD

**Issue by BenjaminHamon** [PyCQA/pylint#3551]
This issue is somewhat convoluted, so bear with me.

I have a python virtual environment (venv) in which I installed packages from their wheel distribution, and from where I run a python application. This application invokes, as a subprocess, a python from a different, unrelated, venv, where the same packages are installed but with the --editable option. In this context, the pylint installed in the second venv raises wrong-import-order warnings because a package is detected as third party when it should be detected as first party. Removing the package from the first venv fixes the issue, making it appear that pylint somehow check with the package installed in the first venv (which it should not know about). The issue only occurs when the first python is used activated (instead of executed directly from its path) and when VIRTUAL_ENV is set.

For context, this is a CI runner which makes a local git clone of itself (the CI project) and run pylint on it.

Here is the output of commands and pylint:

```
++ /usr/bin/python3 -m venv venv_outer
++ venv_outer/bin/python -m pip install --quiet --upgrade pip
++ venv_outer/bin/python -m pip install --quiet ./my_package_root/dist/my_package-0.0.0-py3-none-any.whl
++ /usr/bin/python3 -m venv venv_local
++ venv_local/bin/python -m pip install --quiet --upgrade pip
++ venv_local/bin/python -m pip install --quiet --upgrade pylint
++ venv_local/bin/python -m pip install --quiet --editable my_package_root
++ venv_local/bin/python -m pip install --quiet --editable my_package_other_root
++ venv_outer/bin/python -c 'import subprocess; subprocess.call([ '\''venv_local/bin/python'\'', '\''-c'\'', '\''import sys; print(sys.executable)'\'' ])'
/home/benjamin/debug_2/venv_local/bin/python
++ venv_outer/bin/python -c 'import subprocess; subprocess.call([ '\''venv_local/bin/python'\'', '\''-m'\'', '\''pylint'\'', '\''--score'\'', '\''no'\'', '\''my_package_other'\'' ])'
++ set +x
+ source venv_outer/bin/activate
++ python -c 'import subprocess; subprocess.call([ '\''venv_local/bin/python'\'', '\''-c'\'', '\''import sys; print(sys.executable)'\'' ])'
/home/benjamin/debug_2/venv_local/bin/python
++ python -c 'import subprocess; subprocess.call([ '\''venv_local/bin/python'\'', '\''-m'\'', '\''pylint'\'', '\''--score'\'', '\''no'\'', '\''my_package_other'\'' ])'
************* Module my_package_other.my_module
my_package_other_root/my_package_other/my_module.py:2:0: C0411: third party import "import my_package" should be placed before "import my_package_other" (wrong-import-order)
++ set +x
+ deactivate
```

### Steps to reproduce

Download [pylint_debug.zip](https://github.com/PyCQA/pylint/files/4561498/pylint_debug.zip), extract it and run debug.sh.

Or manually:

- Create two packages, the second one with a module importing its own package, then the first package
- Build a distribution package for the first package
- Create a first virtualenv, venv_outer
- Install the first package distribution in venv_outer
- Create a second virtualenv, venv_local
- Install pylint in venv_local
- Install both packages as editable packages in venv_local
- Run the venv_outer python without activate, invoke as a subprocess the venv_local pylint on the second package
- Notice that pylint report no errors
- Activate the venv_outer python, run the venv_outer python, invoke as a subprocess the venv_local pylint on the second package
- Notice that pylint report errors about wrong-import-order, detecting the first package as being third-party, and the second as first-party
- Activate the venv_outer python, unset VIRTUAL_ENV, run the venv_outer python, invoke as a subprocess the venv_local pylint on the second package
- Notice that pylint report no errors

### Current behavior

Pylint raises wrong-import-order warnings, where packages are wrongly detected as third party.
Behavior is also inconsistent depending on how pylint is invoked.

### Expected behavior

Pylint does not raise wrong-import-order warnings.

### pylint --version output

pylint 2.5.0
astroid 2.4.0
Python 3.5.3 (default, Sep 27 2018, 17:25:39)
[GCC 6.3.0 20170516]

(On a Debian 9)

Tried quickly to reproduce on Windows 10 with Python 3.8.1, the issue did not occur.

**Comment by Pierre-Sassoulas** [comment]
I have the same issue using `isort` with `pre-commit`. I fix it by having an isort.cfg with `known_third_parties` set correctly. For pylint you can do : 
```
[IMPORTS]:

# Force import order to recognize a module as part of a third-party library.
known-third-party=your modules
```

I'm not saying that it's not something that isort/pylint could do better, but here's a quicker fix to your problem. :)

**Comment by BenjaminHamon** [comment]
That's good to know, thanks.

Once I had found out when the issue occurred, I worked around it by simply not using activate. What worries me is it looks like packages installed in another venv affect pylint results.

**Comment by zachliu** [comment]
i'm having a similar issue with both `pylint` and `isort`, i had to do 
```
[IMPORTS]

# Force import order to recognize a module as part of a third-party library.
known-third-party=airflow
```
in my `.pylintrc` and
```
isort --check-only --diff --thirdparty airflow <my_directory>
```
for my `isort`

**Comment by Pierre-Sassoulas** [comment]
The check inside pylint is literally the same than in isort (pylint's use isort internally), if you're using isort you can disable `wrong-import-order` in your pylintrc:

```ini
[MASTER]

disable=wrong-import-order
```
